src/kudu/consensus/consensus_queue.cc

```diff
@@ -688,6 +690,13 @@ Status PeerMessageQueue::RequestForPeer(const string& uuid,
   }
   *needs_tablet_copy = false;
 
+  // If the next hop != the destination, we are sending these messages via a proxy.
+  string next_hop_uuid = routing_table_.NextHop(local_peer_pb_.permanent_uuid(), uuid);
```

Comment:
seems like lock is not held here.

Reply:
Good catch! We already took and released the lock in this function, so I'll just move the call up there.

Reply:
Fixed